### PR TITLE
Bug 1437384 - phabbugz_feed.pl in PhabBugz extension should be extended to also query for new users in Phab

### PR DESCRIPTION
**Pull request by dklawren**

### CODE REVIEW

File: extensions/PhabBugz/lib/Feed.pm [hunk 1]
```diff
@@ -347,21 +360,148 @@ sub process_revision_change {
...
-    $self->logger->info("SUCCESS");
+    $self->logger->info('SUCCESS: Revision D' . $revision->id . ' processed');
 }
 
-sub feed_transactions {
-    my ($self, $after) = @_;
+sub process_new_user {
+    my ( $self, $object_phid ) = @_;
+
+    my $phab_user = Bugzilla::Extension::PhabBugz::User->new_from_query( { phids => [ $object_phid ] } );
```
Comment (zalun) [inline extensions/PhabBugz/lib/Feed.pm]: `User->new_from_query` is requesting Phabricator to get user's data. Can't we use the data already requested in the `new_users` method?

File: extensions/PhabBugz/lib/Revision.pm [hunk 2]
```diff
@@ -9,6 +9,7 @@ package Bugzilla::Extension::PhabBugz::Revision;
 
 use 5.10.1;
 use Moo;
+use Scalar::Util qw(blessed);
```
[the remark this answers is not in the document]
Comment (dklawren) [reply extensions/PhabBugz/lib/Revision.pm]: Fix an error occurring in Revision.pm

File: extensions/PhabBugz/lib/Feed.pm (hunk 1 above)
Comment (dklawren) [reply extensions/PhabBugz/lib/Feed.pm]: I can try that and see. If we pass in the user data to new() then it should convert it to an object the same as if we use new_from_query().

File: extensions/PhabBugz/lib/Revision.pm (hunk 2 above)
[the remark this answers is not in the document]
Comment (zalun) [reply extensions/PhabBugz/lib/Revision.pm]: I don't know why, but it's passing in review 419 with `use Scalar::Util 'blessed';`

[the remark this answers is not in the document]
Comment (dklawren) [reply extensions/PhabBugz/lib/Revision.pm]: They are the same. I prefer qw().

**Comment by dylanwh** [comment]
Note that #423 has been added (bug-pending) that will provide better diagnostics the next time someone encounters the error you found.

**Pull request merged**